### PR TITLE
Code cleanup: Use Java17 features

### PR DESCRIPTION
**Pull request by holgerfriedrich**
Use new language features, mainly instanceof pattern matching.
This change covers almost all of the bundles, the itests are still to be done in another PR.
Relates to #3508.

### CODE REVIEW

**Comment by holgerfriedrich** [comment]
Sorry, I had to rebase due to merge conflict.

@wborn would you be willing to review this? Just asking because you did the review of #3508 ;-)
 Should I split the PR up in smaller chunks?

**Comment by wborn** [comment]
Thanks for all the cleanup @holgerfriedrich! :+1: I can have a look at the PR like it is now. :slightly_smiling_face: Because this PR touches a lot of code, it's probably better to only merge it after M2 is released on Sunday.

**Comment by holgerfriedrich** [comment]
Good approach, thanks.
Is there a place where developers can find the planned release schedule?

**Comment by wborn** [comment]
Milestones are usually released in the weekend at the beginning of each month. But they can get postponed in case of stability issues or when maintainers are not available.

Patch releases are usually made on demand when there is enough content or some critical issue (like log4j) pops up.

**Review by wborn — CHANGES_REQUESTED** [review]
It was a big scroll but all the cleanup looks very good to me! :slightly_smiling_face: 
I only have a few minor comments below:

File: bundles/org.openhab.core/src/main/java/org/openhab/core/items/Item.java [hunk 1]
```diff
@@ -161,7 +161,7 @@ public interface Item extends Identifiable<String> {
      *
      * @return the {@link CommandDescription} for the default locale (can be null).
      */
-    public default @Nullable CommandDescription getCommandDescription() {
+    default public @Nullable CommandDescription getCommandDescription() {
```
[the remark this answers is not in the document]
Comment (J-N-K) [reply bundles/org.openhab.core/src/main/java/org/openhab/core/items/Item.java:164]: I would suggest to omit `public`. Methods on interfaces are always public.

[the remark this answers is not in the document]
Comment (wborn) [reply bundles/org.openhab.core/src/main/java/org/openhab/core/items/Item.java:164]: Yes that's true! Just remove all the `public` modifiers from the interface. :slightly_smiling_face:

[the remark this answers is not in the document]
Comment (holgerfriedrich) [reply bundles/org.openhab.core/src/main/java/org/openhab/core/items/Item.java]: for now just reverted to status before - lets do this together with the missing parts

**Comment by holgerfriedrich** [comment]
sorrry had to force push to resolve merge conflict with current main (any way better than fetch upstream, rebase upstream/main, push --force?)

**Review by wborn — APPROVED** [review]
LGTM now!

**Comment by wborn** [comment]
> any way better than fetch upstream, rebase upstream/main, push --force?

Not really, except for using `--force-with-lease` instead, but that's a bit more chars to type. 😉

**Pull request merged**